geo-types/src/rect.rs

```diff
@@ -34,11 +34,23 @@ impl<T: CoordinateType> Rect<T> {
     where
         C: Into<Coordinate<T>>,
     {
-        let (min, max) = (min.into(), max.into());
-
-        Self::assert_valid_bounds(min, max);
+        Rect::try_new(min, max).expect(RECT_INVALID_BOUNDS_ERROR)
+    }
 
-        Rect { min, max }
+    pub fn try_new<C>(min: C, max: C) -> Option<Rect<T>>
```

Comment:
Take it or leave it, but would this make more sense as a Result?

Reply:
https://github.com/georust/geo/pull/442/commits/da09297b9ab787423f9649063a223359fc73c413